src/economy/ferry_fleet.cc

```diff
@@ -84,6 +84,14 @@ bool FerryFleet::init(EditorGameBase& egbase) {
 	return find_other_fleet(egbase);
 }
 
+bool FerryFleet::init(EditorGameBase& egbase, Waterway* waterway) {
```

Comment:
What is the purpose of this change?

Reply:
`Waterway::request_ferry()` called `FerryFleet::request_ferry()` before `FerryFleet::init()`. This caused trouble when I switched `Waterway:fleet_` to `OPtr`, because it was being assigned a `FerryFleet` without a serial.